File: cookbooks/agent-knowledge-base/02-load.py
"""Step 02 — Load data/corpus.json into ArcFlow.

Run:
    uv run python 02-load.py
"""
from __future__ import annotations

import time

from _load import load


def main() -> None:
    t0 = time.perf_counter()
    db, payload = load(verbose=True)
    elapsed = time.perf_counter() - t0
    n_items = (
        len(payload["documents"])
        + len(payload["entities"])
        + len(payload["mentions"])
        + len(payload["relations"])
    )
    print(f"loaded in {elapsed:.2f}s ({n_items / elapsed:.0f} items/sec)")

    n_docs = db.execute("MATCH (d:Document) RETURN count(d)").get(0, 0)
    n_entities = db.execute("MATCH (e:Entity) RETURN count(e)").get(0, 0)
    n_mentions = db.execute("MATCH (m:Mention) RETURN count(m)").get(0, 0)
    n_relations = db.execute("MATCH (r:Relation) RETURN count(r)").get(0, 0)
    n_has = db.execute("MATCH ()-[r:HAS]->() RETURN count(r)").get(0, 0)
    n_cites = db.execute("MATCH ()-[r:CITES]->() RETURN count(r)").get(0, 0)

    print(
        f"in-graph: documents={n_docs}, entities={n_entities}, "
        f"mentions={n_mentions}, relations={n_relations}"
    )
    print(f"          HAS edges={n_has}, CITES edges={n_cites}")
    db.close()
    print("OK")


if __name__ == "__main__":
    main()

File: examples/agent-knowledge-base/rust/src/bin/02-load.rs
//! Step 02 (Rust) — load the corpus into ArcFlow.
//!
//! Mirrors `02-load.py`. Reads `../data/corpus.json`, builds the four-
//! label GraphRAG schema (Document / Entity / Mention / Relation), and
//! prints summary counts.

use agent_knowledge_base::load;
use anyhow::Result;
use std::path::PathBuf;

fn main() -> Result<()> {
    let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
    path.pop(); // rust/ → cookbook root
    path.push("data");
    path.push("corpus.json");

    let (db, corpus) = load(&path)?;

    println!(
        "loaded {} docs, {} entities, {} mentions, {} relations",
        corpus.documents.len(),
        corpus.entities.len(),
        corpus.mentions.len(),
        corpus.relations.len(),
    );

    // Sanity check: count nodes via query — same result as the Python loader.
    for (label, q) in [
        ("Document", "MATCH (d:Document) RETURN count(*) AS n"),
        ("Entity",   "MATCH (e:Entity)   RETURN count(*) AS n"),
        ("Mention",  "MATCH (m:Mention)  RETURN count(*) AS n"),
        ("Relation", "MATCH (r:Relation) RETURN count(*) AS n"),
    ] {
        let result = db.execute(q)?;
        let n = result.rows.first()
            .and_then(|row| row.get("n"))
            .cloned()
            .unwrap_or_else(|| "0".to_string());
        println!("  {label:<10} {n}");
    }

    Ok(())
}
